tavern/response.py

```diff
@@ -131,7 +131,9 @@ def check_ext_functions(verify_block):
                     "Badly formatted 'verify_response_with' block"
                 )
 
-        check_ext_functions(response_block.get("verify_response_with", None))
+        mqtt_responses = response_block.get("mqtt_responses")
+        for mqtt_response in mqtt_responses:
+            check_ext_functions(mqtt_response.get("verify_response_with", None))
```

Comment:
This won't work because the response block can still be just a single mqtt response, or a rest response, or (soon) a gRPC response.

Reply:
As far as I tried with mqtt is working, with single or multiple responses, as even single responses reach this method in the json array form (with only one item, of course). You are totally right about rest response, simply I'm not using rest so I did not think about that.
No idea about future gRPC implementation, so no words about that.

I can try to update pull request with a proper rest response handling if that is enough, but I have no way to make a gRPC-future proof implementation. I was simply trying to fix the bug I reported some days ago.

Reply:
You don't need to worry about the grpc thing, I'll handle that in the appropriate PR, I mainly just meant it still needs to work for rest responses. Something like
```python
if mqtt_responses := response_block.get("mqtt_responses"):
  for mqtt_response in mqtt_responses:
    check_ext_functions(mqtt_response.get("verify_response_with", None))
else:
  check_ext_functions(response_block.get("verify_response_with", None))
```

Reply:
I have updated PR with your code suggestion, thank you!